deployments/pulse3d/services/pulse3d/src/main.py

```diff
@@ -439,6 +439,17 @@ async def create_new_job(
             analysis_params[param] = _format_tuple_param(analysis_params[param], default_values)
 
         logger.info(f"Using v{details.version} with params: {analysis_params}")
+        # check if pulse3d version is available
+        pulse3d_version_status = await con.fetchrow(
+            "SELECT state, end_of_life_date FROM pulse3d_versions WHERE version = $1", details.version
+        )
+        if (
```

Comment:
Is there any reason the date isn't checked inside the query? Also what happens when the date is `null` in the database? I think `strptime` would error in that situation
```
>>> datetime.datetime.strptime(None, "%Y-%m-%d")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: strptime() argument 1 must be str, not None
```